diff -r 120f9bb88318 imagekit/specs.py
--- a/imagekit/specs.py	Sat Dec 19 11:01:54 2009 -0500
+++ b/imagekit/specs.py	Mon Jan 11 12:09:12 2010 -0600
@@ -73,21 +73,22 @@
 
     @property
     def name(self):
-        filepath, basename = os.path.split(self._obj._imgfield.name)
-        filename, extension = os.path.splitext(basename)
-        for processor in self.spec.processors:
-            if issubclass(processor, processors.Format):
-                extension = processor.extension
-        cache_filename = self._obj._ik.cache_filename_format % \
-            {'filename': filename,
-             'specname': self.spec.name(),
-             'extension': extension.lstrip('.')}
-        if callable(self._obj._ik.cache_dir):
-            return self._obj._ik.cache_dir(self._obj, filepath,
-                                           cache_filename)
-        else:
-            return os.path.join(self._obj._ik.cache_dir, filepath,
-                                cache_filename)
+        if self._obj._imgfield.name:
+            filepath, basename = os.path.split(self._obj._imgfield.name)
+            filename, extension = os.path.splitext(basename)
+            for processor in self.spec.processors:
+                if issubclass(processor, processors.Format):
+                    extension = processor.extension
+            cache_filename = self._obj._ik.cache_filename_format % \
+                {'filename': filename,
+                 'specname': self.spec.name(),
+                 'extension': extension.lstrip('.')}
+            if callable(self._obj._ik.cache_dir):
+                return self._obj._ik.cache_dir(self._obj, filepath,
+                                               cache_filename)
+            else:
+                return os.path.join(self._obj._ik.cache_dir, filepath,
+                                    cache_filename)
 
     @property
     def url(self):
